pkg/controllers/mcs/service_export_controller.go

```diff
@@ -139,6 +143,52 @@ func (c *ServiceExportController) RunWorkQueue() {
 	}
 	c.worker = util.NewAsyncWorker(workerOptions)
 	c.worker.Run(c.WorkerNumber, c.StopChan)
+
+	go c.enqueueReportedEpsServiceExport()
+}
+
+func (c *ServiceExportController) enqueueReportedEpsServiceExport() {
+	workList := &workv1alpha1.WorkList{}
+	err := wait.PollUntilContextCancel(context.TODO(), time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
There is [an issue](https://github.com/kubernetes/kubernetes/pull/119762) with the `wait.PollUntilContextCancel`, that makes the condition will be executed twice immediately.

This issue has been fixed in Kubernetes v1.29 but we haven't adopted it. We can continue to use the deprecated functions for now. This deprecation work is tracked by #3835.

Reply:
Ok, thanks, let me update it to wait.Poll first.

Reply:
Update it to the `wait.PollUntil`, please help take a review again.